spec/factories/users.rb

```diff
@@ -32,8 +32,75 @@
     last_name { Faker::Name.last_name }
     sequence(:email) { |index| Faker::Internet.email.sub(/\@/, "_#{index}@") }
     password { Faker::Internet.password }
-    role { Faker::Job.title }
+    role { "user" }
     department { Faker::Job.field }
     about { Faker::Lorem.sentences }
   end
-end
+
+  trait :admin do
+    role { "admin" }
+  end
+
+  trait :with_workspaces do
+    after(:create) do |user|
+      FactoryBot.create_list(:workspace, 2, user: user)
+    end
+  end
+
+  trait :with_projects do
+    after(:create) do |user|
+      user.workspaces.each do |workspace|
+        FactoryBot.create_list(:project, 2, workspace: workspace, users: [user])
+      end
+    end
+  end
+
+  trait :comments_for_projects do
+    after(:create) do |user|
+      user.workspaces.each do |workspace|
+        workspace.projects.each do |project|
+          project.update(users: [user])
+          FactoryBot.create_list(:comment, 1.upto(5).to_a.sample, :for_project, user: user, commentable: project)
```

Comment:
this code 1.upto(5).to_a.sample is bad
rework

Reply:
for the traits - add a controllable static value of number of comments, could be very useful later.. 
tip: `transient + evaluator`